ssl/quic/quic_impl.c

```diff
@@ -2287,9 +2289,15 @@ static size_t ossl_quic_pending_int(const SSL *s, int check_channel)
     size_t avail = 0;
     int fin = 0;
 
-    if (!expect_quic_with_stream_lock(s, /*remote_init=*/-1, &ctx))
+
```

Comment:
Can you explain the rationale behind this change?

Reply:
s_client calls `SSL_has_pending()` very early - even before a stream has been associated with the SSL object (during handshake). The TLS implementation just returns 0 in this case (i.e. no data pending), but `expect_quic_with_stream_lock` fails if there is no stream raising a non-normal error and causing subsequent calls to `SSL_get_error()` to return `SSL_ERROR_SSL`.

Reply:
OK, makes sense.

Reply:
Hmm... how does SSL_read()/SSL_write() work in this regard?

Reply:
Looking at the source it seems SSL_read() should work however SSL_write() doesn't as ossl_quic_write() calls expect_quic_with_stream_lock() at the beginning.